ui/webui/src/components/app.jsx

```diff
@@ -127,31 +127,34 @@ export const Application = () => {
                       exception={{ ...criticalError, jsMessage: jsError?.message, backendMessage: criticalError?.message, stack: jsError?.stack }}
                       isConnected={state.network.connected}
                       reportLinkURL={bzReportURL} />}
-                <Page
-                  data-debug={conf.Anaconda.debug}
-                >
-                    <PageGroup stickyOnBreakpoint={{ default: "top" }}>
-                        <AnacondaHeader
-                          title={title}
-                          reportLinkURL={bzReportURL}
-                          isConnected={state.network.connected}
-                          onCritFail={onCritFail}
-                        />
-                    </PageGroup>
-                    <AddressContext.Provider value={address}>
-                        <WithDialogs>
-                            <AnacondaWizard
-                              onCritFail={onCritFail}
-                              title={title}
-                              storageData={state.storage}
-                              localizationData={state.localization}
-                              dispatch={dispatch}
-                              conf={conf}
-                              osRelease={osRelease}
-                            />
-                        </WithDialogs>
-                    </AddressContext.Provider>
-                </Page>
+                {!jsError
+                    ? (
+                        <Page
+                          data-debug={conf.Anaconda.debug}
+                        >
+                            <PageGroup stickyOnBreakpoint={{ default: "top" }}>
+                                <AnacondaHeader
+                                  title={title}
+                                  reportLinkURL={bzReportURL}
+                                  isConnected={state.network.connected}
+                                  onCritFail={onCritFail}
+                                />
+                            </PageGroup>
+                            <AddressContext.Provider value={address}>
+                                <WithDialogs>
+                                    <AnacondaWizard
+                                      onCritFail={onCritFail}
+                                      title={title}
+                                      storageData={state.storage}
+                                      localizationData={state.localization}
+                                      dispatch={dispatch}
+                                      conf={conf}
+                                      osRelease={osRelease}
+                                    />
+                                </WithDialogs>
+                            </AddressContext.Provider>
+                        </Page>)
+                    : (<Page>{null}</Page>)}
```

Comment:
This looks pretty weird. Please put the Page wrapper to have child the 'Critical Error' component. Then put this condition to the rest of the children.